### ISSUE TITLE
Having Slyguy Common installed causes Kodi to break

### ISSUE THREAD

**Issue by Neriderc** [matthuisman/slyguy.addons#804]
In the last couple of days (possibly after an update?) all my "SlyGuy" addons don't work. SkyGo, Disney+, etc.

If I try to exit Kodi, it freezes and never recovers (left it for hours). I worked out if I disable all addons marked as SlyGuy then Kodi works again, and I can exit fine. If I only enable "SlyGuy Common", it freezes on exit again.

Here are the logs from the shutdown sequence. Nothing stands out except that a slyguy script that didn't run in time and was killed. There are no actual crash logs, just this in `.kodi/temp/kodi.log`.

```
2024-07-20 23:08:38.709 T:36820    INFO <general>: Stopping player
2024-07-20 23:08:38.709 T:36820    INFO <general>: Storing total System Uptime
2024-07-20 23:08:38.709 T:36820    INFO <general>: Saving settings
2024-07-20 23:08:38.714 T:36820    INFO <general>: Saving skin settings
2024-07-20 23:08:38.715 T:36820    INFO <general>: Stopping all
2024-07-20 23:08:38.715 T:36820    INFO <general>: ES: Stopping event server
2024-07-20 23:08:38.715 T:36820    INFO <general>: stopping zeroconf publishing
2024-07-20 23:08:38.727 T:36854    INFO <general>: WebSocket: connection closed by client
2024-07-20 23:08:38.727 T:36854    INFO <general>: JSONRPC Server: Disconnection detected
2024-07-20 23:08:38.750 T:36820    INFO <CWebserver[8080]>: Stopped
2024-07-20 23:08:39.703 T:36853    INFO <general>: ES: UDP Event server stopped
2024-07-20 23:08:39.703 T:36820    INFO <general>: stop dvd detect media
2024-07-20 23:08:39.704 T:36820    INFO <general>: CServiceAddonManager: failed to stop inputstream.ffmpegdirect (may have ended)
2024-07-20 23:08:39.704 T:36847    INFO <general>: CPythonInvoker(1, /home/username/.kodi/addons/script.logviewer/service.py): script successfully run
2024-07-20 23:08:39.722 T:36847    INFO <general>: Python interpreter interrupted by user
2024-07-20 23:08:39.867 T:36848    INFO <general>: CPythonInvoker(2, /home/username/.kodi/addons/script.module.slyguy/service.py): script successfully run
2024-07-20 23:08:44.728 T:36820   ERROR <general>: CPythonInvoker(2, /home/username/.kodi/addons/script.module.slyguy/service.py): script didn't stop in 5 seconds - let's kill it
```

Here's some maybe relevant stuff as well?

```
2024-07-20 23:17:44.021 T:38333    INFO <general>: script.module.slyguy - Connecting to db: /home/username/.kodi/userdata/addon_data/script.module.slyguy/settings.db
2024-07-20 23:17:44.023 T:38333    INFO <general>: script.module.slyguy - Using new settings system
2024-07-20 23:17:44.169 T:38333    INFO <general>: script.module.slyguy - Python Version: 3.10.12 (main, Mar 22 2024, 16:50:05) [GCC 11.4.0]
2024-07-20 23:17:44.170 T:38333    INFO <general>: script.module.slyguy - Proxy Started: http://127.0.0.1:44575/
2024-07-20 23:17:44.171 T:38333    INFO <general>: script.module.slyguy - Widevine Level (fallback): 3
2024-07-20 23:17:44.171 T:38333    INFO <general>: script.module.slyguy - HDCP Level (fallback): 0.0
```

I don't even seem to be able to uninstall the addon. If I try it freezes as well.

What should I be doing to troubleshoot this issue?

**Comment by matthuisman** [comment]
What version Kodi and what device?

**Comment by Neriderc** [comment]
It's running on Ubuntu on an old laptop. Thanks for the prompt to check the version, it's running a 2 year out of date version installed via Ubuntu repository. I'll update and report back, but it's version 19 and the latest seems to be 21 so I'm guessing this is the problem.

**Comment by matthuisman** [comment]
no. that wont be it. just curious.
I suspect its because i wasnt closing the DB connection when the service stops.
I've just pushed update slyguy.common that hopefully fixes the issue. Please update and test

**Comment by matthuisman** [comment]
that should solve the shutdown hang.

Did you say you cant use any of the scripts though?
Any errors around that?

**Comment by 3DSpotters** [comment]
Hi I have kind of same issue: SlyGuy Common v.0.84.1.
I can not access SETTINGS. They are greyed.
The problem is my MAX addon do not show any movies or series...

**Comment by Neriderc** [comment]
I'm now on v0.84.2 and the shutdown hang is still there, same error.

Here's some logs for Disney+. I can open it and get to the first screen with options like "Featured", "Hubs", "Movies", "Series". For this I clicked "Featured", and it stays with the loading circle.

Here's the logs:
```
2024-07-21 10:44:35.700 T:129824    INFO <general>: Loading skin file: MyVideoNav.xml, load type: KEEP_IN_MEMORY
2024-07-21 10:44:35.770 T:130009    INFO <general>: initializing python engine.
2024-07-21 10:44:36.541 T:130009    INFO <general>: slyguy.disney.plus - Connecting to db: /home/username/.kodi/userdata/addon_data/script.module.slyguy/settings.db
2024-07-21 10:44:36.549 T:130009    INFO <general>: slyguy.disney.plus - Closing db: /home/username/.kodi/userdata/addon_data/script.module.slyguy/settings.db
2024-07-21 10:44:37.956 T:130014    INFO <general>: initializing python engine.
2024-07-21 10:44:56.858 T:129861   ERROR <general>: CCurlFile::Stat - Failed: Timeout was reached(28) for http://127.0.0.1:44575/https://prod-ripcut-delivery.disney-plus.net/v1/variant/disney/5FDFAFEDDDBC44CF343658D8EC2E4594A0AD0E3338B3C20FD87892F57347E188/scale?width=300|user-agent=okhttp/4.9.3&session_type=art&session_addonid=slyguy.disney.plus
```

**Comment by matthuisman** [comment]
Thanks. I'll get my Ubuntu up and test

**Comment by Neriderc** [comment]
As an update, I have it working by updating to the flatpack v21 of Kodi. No hang on exit, and so far I've tested Disney+ and Sky Go and they are working fine again. 

I uninstalled the old version so I probably can't help with troubleshooting if you're still keen on testing Kodi 19, I'll leave this open in case you were still wanting to track it but feel free to close if you want as I have my setup working again.
